src/main/webapp/app/shared/components/code-button/code-button.component.ts

```diff
@@ -79,27 +80,32 @@ export class CodeButtonComponent implements OnInit, OnChanges {
         private ideSettingsService: IdeSettingsService,
     ) {}
 
-    ngOnInit() {
-        this.accountService
-            .identity()
-            .then((user) => {
-                this.user = user!;
-                this.refreshTokenState();
+    async ngOnInit() {
+        const user = await this.accountService.identity();
 
-                this.copyEnabled = true;
-                this.useSsh = this.localStorage.retrieve('useSsh') || false;
-                this.useToken = this.localStorage.retrieve('useToken') || false;
-                this.localStorage.observe('useSsh').subscribe((useSsh) => (this.useSsh = useSsh || false));
-                this.localStorage.observe('useToken').subscribe((useToken) => (this.useToken = useToken || false));
+        this.user = user!;
```

Comment:
You could also do

`this.user = await this.accountService.identity()!;`

since the local `user` variable doesn't seem to be used anywhere

Reply:
good point, that makes it a bit more concise :+1: